### PR TITLE
Fix crash-on-launch when post list is empty

### LINKED ISSUE #235: App crashes if launched with empty post list
I think this is what we're seeing in [this forum topic](https://discuss.write.as/t/ios-app-crashes-at-launch/7315/8): on the first launch after installing, the app creates a new draft, navigates you to it, then navigates back to the Drafts list (this is a regression!). Because the post is empty, it's deleted.

Launching the app a second time does the same thing, except once it navigates back to the post list, it crashes.

```
Incident Identifier: 781E98F0-CD62-4589-B563-8C7911E1B8D5
Hardware Model:      iPhone13,3
Process:             WriteFreely-MultiPlatform [5198]
Path:                /private/var/containers/Bundle/Application/AB4BF378-997F-4821-9228-1595A5A909BE/WriteFreely-MultiPlatform.app/WriteFreely-MultiPlatform
Identifier:          com.abunchtell.WriteFreely-MultiPlatform
Version:             1.0.14 (690)
AppStoreTools:       14B44
AppVariant:          1:iPhone13,3:16
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           com.abunchtell.WriteFreely-MultiPlatform [1592]

Date/Time:           2022-12-04 07:22:58.8109 -0500
Launch Time:         2022-12-04 07:22:57.3830 -0500
OS Version:          iPhone OS 16.1.2 (20B110)
Release Type:        User
Baseband Version:    3.07.00
Report Version:      104

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x000000019eb4c7cc
Termination Reason: SIGNAL 5 Trace/BPT trap: 5
Terminating Process: exc handler [5198]

Triggered by Thread:  0


Thread 0 name:
Thread 0 Crashed:
0   SwiftUI                       	0x000000019eb4c7cc specialized Array._checkSubscript(_:wasNativeTypeChecked:) + 32 (<compiler-generated>:0)
1   SwiftUI                       	0x000000019ece8e34 Sections.startOffset(forSectionAt:) + 40 (Sections.swift:116)
2   SwiftUI                       	0x000000019ed254dc Sections.rowIDs(forSectionAt:) + 36 (Sections.swift:140)
3   SwiftUI                       	0x000000019edc70ec CollectionViewListDataSource.rowIDs(forSectionAt:) + 32 (CollectionViewListDataSource.swift:129)
4   SwiftUI                       	0x000000019edc7be4 protocol witness for ListDiffable.rowIDs(forSectionAt:) in conformance CollectionViewListDataSource<A> + 36 (<compiler-generated>:0)
5   SwiftUI                       	0x000000019ebafb00 ShadowRowCollection.init(_:section:) + 972 (ShadowListUpdateRecorder.swift:293)
6   SwiftUI                       	0x000000019ebdd0f0 ShadowListUpdateRecorder.rowIDs(forSectionAt:) + 320 (ShadowListUpdateRecorder.swift:250)
7   SwiftUI                       	0x000000019ffac8c0 ListDiffable.rowIndex(at:) + 352 (ListDiffable.swift:70)
8   SwiftUI                       	0x000000019ebe7da4 ListCoreDataSource.visitContent<A>(atRow:visitor:) + 408 (ListCoreDataSource.swift:428)
9   SwiftUI                       	0x000000019f931170 UICollectionViewListCoordinatorBase.updateListContents(_:) + 896 (UICollectionViewListCoordinator.swift:1124)
10  SwiftUI                       	0x000000019f933430 closure #3 in performUpdates #1 <A, B>() in UICollectionViewListCoordinatorBase.update(_:to:transaction:performDiff:) + 44 (UICollectionViewListCoordinator.swift:982)
11  SwiftUI                       	0x000000019ec0bef8 thunk for @callee_guaranteed () -> () + 28 (<compiler-generated>:0)
12  SwiftUI                       	0x000000019ed56fa0 thunk for @escaping @callee_guaranteed () -> () + 28 (<compiler-generated>:0)
13  UIKitCore                     	0x000000019d392564 +[UIView(Animation) performWithoutAnimation:] + 76 (UIView.m:14722)
14  SwiftUI                       	0x000000019f931bcc performUpdates #1 <A, B>() in UICollectionViewListCoordinatorBase.update(_:to:transaction:performDiff:) + 1220 (UICollectionViewListCoordinator.swift:977)
15  SwiftUI                       	0x000000019ec0bef8 thunk for @callee_guaranteed () -> () + 28 (<compiler-generated>:0)
16  SwiftUI                       	0x000000019ed56fa0 thunk for @escaping @callee_guaranteed () -> () + 28 (<compiler-generated>:0)
17  UIKitCore                     	0x000000019d392564 +[UIView(Animation) performWithoutAnimation:] + 76 (UIView.m:14722)
18  SwiftUI                       	0x000000019f933598 closure #2 in UICollectionViewListCoordinatorBase.update(_:to:transaction:performDiff:) + 348 (UICollectionViewListCoordinator.swift:1007)
19  SwiftUI                       	0x000000019ec27e30 thunk for @escaping @callee_guaranteed () -> () + 28 (<compiler-generated>:0)
20  SwiftUI                       	0x000000019f5597fc static Update.dispatchActions() + 572 (<compiler-generated>:0)
21  SwiftUI                       	0x000000019eb58850 static Update.end() + 116 (Update.swift:58)
22  SwiftUI                       	0x000000019eb4b1c0 static NSRunLoop.flushObservers() + 164 (StandardLibraryAdditions.swift:852)
23  SwiftUI                       	0x000000019eb4b0d4 closure #1 in static NSRunLoop.addObserver(_:) + 52 (StandardLibraryAdditions.swift:829)
24  SwiftUI                       	0x000000019eb4b368 @objc closure #1 in static NSRunLoop.addObserver(_:) + 60 (<compiler-generated>:0)
25  CoreFoundation                	0x000000019b235d60 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 36 (CFRunLoop.c:1789)
26  CoreFoundation                	0x000000019b1be2d0 __CFRunLoopDoObservers + 532 (CFRunLoop.c:1902)
27  CoreFoundation                	0x000000019b21fc7c __CFRunLoopRun + 1048 (CFRunLoop.c:2981)
28  CoreFoundation                	0x000000019b224ed4 CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
29  GraphicsServices              	0x00000001d4526368 GSEventRunModal + 164 (GSEvent.c:2196)
30  UIKitCore                     	0x000000019d7033d0 -[UIApplication _run] + 888 (UIApplication.m:3745)
31  UIKitCore                     	0x000000019d703034 UIApplicationMain + 340 (UIApplication.m:5335)
32  SwiftUI                       	0x000000019ed10014 closure #1 in KitRendererCommon(_:) + 168 (UIKitApp.swift:37)
33  SwiftUI                       	0x000000019ec7116c runApp<A>(_:) + 152 (<compiler-generated>:0)
34  SwiftUI                       	0x000000019ec5a4bc static App.main() + 128 (App.swift:114)
35  WriteFreely-MultiPlatform     	0x0000000102d47f44 static CheckForDebugModifier.main() + 24 (WriteFreely_MultiPlatformApp.swift:20)
36  WriteFreely-MultiPlatform     	0x0000000102d47f44 static CheckForDebugModifier.$main() + 24 (<compiler-generated>:7)
37  WriteFreely-MultiPlatform     	0x0000000102d47f44 main + 36 (ContentView.swift:0)
38  dyld                          	0x00000001b9890960 start + 2528 (dyldMain.cpp:1170)

Thread 1:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 3:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 4:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 5 name:
Thread 5:
0   libsystem_kernel.dylib        	0x00000001d7da5b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	0x00000001d7db8008 mach_msg2_internal + 80 (mach_msg.c:201)
2   libsystem_kernel.dylib        	0x00000001d7db8248 mach_msg_overwrite + 388 (mach_msg.c:0)
3   libsystem_kernel.dylib        	0x00000001d7da608c mach_msg + 24 (mach_msg.c:323)
4   CoreFoundation                	0x000000019b21eaf0 __CFRunLoopServiceMachPort + 160 (CFRunLoop.c:2622)
5   CoreFoundation                	0x000000019b21fd34 __CFRunLoopRun + 1232 (CFRunLoop.c:3005)
6   CoreFoundation                	0x000000019b224ed4 CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
7   Foundation                    	0x00000001955c6334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212 (NSRunLoop.m:373)
8   Foundation                    	0x00000001955c621c -[NSRunLoop(NSRunLoop) runUntilDate:] + 64 (NSRunLoop.m:420)
9   UIKitCore                     	0x000000019d83833c -[UIEventFetcher threadMain] + 436 (UIEventFetcher.m:1377)
10  Foundation                    	0x00000001955df808 __NSThread__start__ + 716 (NSThread.m:963)
11  libsystem_pthread.dylib       	0x00000001e82086cc _pthread_start + 148 (pthread.c:893)
12  libsystem_pthread.dylib       	0x00000001e8207ba4 thread_start + 8

Thread 6:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 7:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 8:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 9:
0   libsystem_pthread.dylib       	0x00000001e8207b90 start_wqthread + 0

Thread 10 name:
Thread 10:
0   libsystem_kernel.dylib        	0x00000001d7da5b48 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	0x00000001d7db8008 mach_msg2_internal + 80 (mach_msg.c:201)
2   libsystem_kernel.dylib        	0x00000001d7db8248 mach_msg_overwrite + 388 (mach_msg.c:0)
3   libsystem_kernel.dylib        	0x00000001d7da608c mach_msg + 24 (mach_msg.c:323)
4   CoreFoundation                	0x000000019b21eaf0 __CFRunLoopServiceMachPort + 160 (CFRunLoop.c:2622)
5   CoreFoundation                	0x000000019b21fd34 __CFRunLoopRun + 1232 (CFRunLoop.c:3005)
6   CoreFoundation                	0x000000019b224ed4 CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
7   Foundation                    	0x00000001955c6334 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212 (NSRunLoop.m:373)
8   Foundation                    	0x00000001955c61c8 -[NSRunLoop(NSRunLoop) run] + 64 (NSRunLoop.m:398)
9   SwiftUI                       	0x000000019ebe9030 static DisplayLink.asyncThread(arg:) + 800 (_UIHostingView.swift:978)
10  SwiftUI                       	0x000000019ebe7168 @objc static DisplayLink.asyncThread(arg:) + 88 (<compiler-generated>:0)
11  Foundation                    	0x00000001955df808 __NSThread__start__ + 716 (NSThread.m:963)
12  libsystem_pthread.dylib       	0x00000001e82086cc _pthread_start + 148 (pthread.c:893)
13  libsystem_pthread.dylib       	0x00000001e8207ba4 thread_start + 8


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000001   x2: 0x00000001f2161a20   x3: 0x000000016d0bd852
    x4: 0x00000001a0234994   x5: 0x0000000000000010   x6: 0x0000000e00000003   x7: 0x0000000000000002
    x8: 0x0000000000000000   x9: 0x0000000129fba6e8  x10: 0x000000019edc7bc0  x11: 0x0000000129fab498
   x12: 0x0000000129fab498  x13: 0x0000000281a14410  x14: 0x01000001f1824b89  x15: 0x00000001f1824b88
   x16: 0x00000002a353925c  x17: 0x323b000129fba6e8  x18: 0x0000000000000000  x19: 0x0000000000000000
   x20: 0x000000016d0bd6b0  x21: 0x00000001f2161a20  x22: 0x000000016d0bd8c0  x23: 0x0000000000000000
   x24: 0x0000000129fbae08  x25: 0x000000016d0bd6b0  x26: 0x0000000129fc4e58  x27: 0x000000016d0bd780
   x28: 0x000000016d0bd7f0   fp: 0x000000016d0bd210   lr: 0x000000019ece8e34
    sp: 0x000000016d0bd180   pc: 0x000000019eb4c7cc cpsr: 0x60001000
   esr: 0xf2000001 (Breakpoint) brk 1


Binary Images:
0x102d40000 - 0x102dbbfff WriteFreely-MultiPlatform arm64  <c060d855496e32a3a40e86ce3048cb21> /private/var/containers/Bundle/Application/AB4BF378-997F-4821-9228-1595A5A909BE/WriteFreely-MultiPlatform.app/WriteFreely-MultiPlatform
0x195584000 - 0x195ecdfff Foundation arm64e  <c431acb6fe043d28b6774de6e1c7d81f> /System/Library/Frameworks/Foundation.framework/Foundation
0x19b1a4000 - 0x19b589fff CoreFoundation arm64e  <5cdc5d9ae5063740b64ebb30867b4f1b> /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation
0x19d361000 - 0x19eb3efff UIKitCore arm64e  <179501b60fc2344ab969b4e3961ebe10> /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore
0x19eb3f000 - 0x1a0482fff SwiftUI arm64e  <494a7d5790af3e9096237038275b8d87> /System/Library/Frameworks/SwiftUI.framework/SwiftUI
0x1b987b000 - 0x1b98fe00f dyld arm64e  <cb3ff411476234d286a4eca13f9fb6c3> /usr/lib/dyld
0x1d4525000 - 0x1d452dfff GraphicsServices arm64e  <a633a095122639428f413877660185ee> /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices
0x1d7da5000 - 0x1d7ddbffb libsystem_kernel.dylib arm64e  <ff27fc8f90ba3332ab7ac5bc2d9ca7b1> /usr/lib/system/libsystem_kernel.dylib
0x1e8207000 - 0x1e8212fff libsystem_pthread.dylib arm64e  <1aa3a4b6f9e730568c8b4e4dd81312a4> /usr/lib/system/libsystem_pthread.dylib

EOF
```

### PR DESCRIPTION
**Pull request by AngeloStavrow**
Closes #235.

### CODE REVIEW

File: Shared/WriteFreely_MultiPlatformApp.swift [hunk 1]
```diff
@@ -152,13 +152,13 @@ struct WriteFreely_MultiPlatformApp: App {
     }
 
     private func showLastDraftOrCreateNewLocalPost() {
-        #if os(macOS)
         if model.editor.lastDraftURL != nil {
-            self.model.selectedPost = model.editor.fetchLastDraftFromAppStorage()
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
```
Comment (AngeloStavrow) [inline Shared/WriteFreely_MultiPlatformApp.swift:156]: While the app is launching, we need to wait a short period of time for `model.selectedPost` to settle before setting it to the last draft, or else it'll get unset.

File: Shared/WriteFreely_MultiPlatformApp.swift [hunk 2]
```diff
@@ -170,7 +170,7 @@ struct WriteFreely_MultiPlatformApp: App {
         let managedPost = model.editor.generateNewLocalPost(withFont: model.preferences.font)
         withAnimation {
             // Set it as the selectedPost
-            DispatchQueue.main.asyncAfter(deadline: .now()) {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
```
Comment (AngeloStavrow) [inline Shared/WriteFreely_MultiPlatformApp.swift:173]: Similarly, wait for the value of `model.selectedPost` to settle before setting it to a new draft.

File: iOS/PostEditor/MultilineTextView.swift [hunk 3]
```diff
@@ -41,7 +41,7 @@ private struct UITextViewWrapper: UIViewRepresentable {
         uiView.font = fontMetrics.scaledFont(for: font)
 
         if uiView.window != nil && isEditing {
-            DispatchQueue.main.async {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.55) {
```
Comment (AngeloStavrow) [inline iOS/PostEditor/MultilineTextView.swift:44]: This delay the janky keyboard-presentation animation when navigating to a post.

| Before | After |
| :---: | :---: |
| <video src="https://user-images.githubusercontent.com/387655/210072805-66d5c83c-9870-4e76-bd58-f3e4926bf7aa.mov" /> | <video src="https://user-images.githubusercontent.com/387655/210072833-17a656de-7d0f-4753-8500-9e8084945e38.mov" /> |

File: iOS/PostEditor/PostTextEditingView.swift [hunk 4]
```diff
@@ -60,8 +60,8 @@ struct PostTextEditingView: View {
                 }
             }
         }
-        .onChange(of: titleIsFirstResponder, perform: { _ in
-            self.bodyIsFirstResponder.toggle()
+        .onChange(of: titleIsFirstResponder, perform: { value in
+            self.bodyIsFirstResponder = !value
```
Comment (AngeloStavrow) [inline iOS/PostEditor/PostTextEditingView.swift:64]: Making explicit that `bodyIsFirstResponder` should be the negation of `titleIsFirstResponder`.

File: iOS/PostEditor/PostTextEditingView.swift [hunk 5]
```diff
@@ -75,6 +75,9 @@ struct PostTextEditingView: View {
             self.titleTextStyle = UIFont(name: appearance.rawValue, size: 26)!
             self.bodyTextStyle = UIFont(name: appearance.rawValue, size: 17)!
         })
+        .onDisappear {
+            hideKeyboard()
+        }
```
Comment (AngeloStavrow) [inline iOS/PostEditor/PostTextEditingView.swift:80]: Fixes the janky keyboard-dismissal animation when leaving the post editor.

**Review by thebaer — APPROVED** [review]
Looks good!

**Pull request merged**